extension/headerssetter/extension.go

```diff
@@ -0,0 +1,122 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package headerssetter // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/headerssetter"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"go.opentelemetry.io/collector/config/configauth"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/headerssetter/source"
+)
+
+type Header struct {
+	key    string
+	source source.Source
+}
+
+func newHeadersSetterExtension(cfg *Config) (configauth.ClientAuthenticator, error) {
+	if cfg == nil {
+		return nil, errors.New("extension configuration is not provided")
+	}
+	if cfg.HeadersConfig == nil {
+		cfg.HeadersConfig = []HeaderConfig{}
```

Comment:
In this case, the extension would still be used and called for every outgoing request, but nothing would be added to the request, right? Would it make sense to return an error here, even though this condition shouldn't happen in the real world, as `Validate` should take care of this?